include/fiction/algorithms/physical_design/apply_gate_library.hpp

```diff
@@ -240,13 +240,13 @@ template <typename CellLyt, typename GateLibrary, typename GateLyt>
  * @tparam CellLyt Type of the returned cell-level layout.
  * @tparam GateLibrary Type of the gate library to apply.
  * @tparam GateLyt Type of the gate-level layout to apply the library to.
- * @tparam Params Type of the parameter used for parameterized gate library.
+ * @tparam Params Type of the parameter used for SiDB on-the-fly gate library.
  * @param lyt The gate-level layout.
  * @param params Parameter for the gate library.
  * @return A cell-level layout that implements `lyt`'s gate types with building blocks defined in `GateLibrary`.
  */
 template <typename CellLyt, typename GateLibrary, typename GateLyt, typename Params>
-[[nodiscard]] CellLyt apply_parameterized_gate_library(const GateLyt& lyt, const Params& params)
+[[nodiscard]] CellLyt apply_sidb_on_the_fly_gate_library(const GateLyt& lyt, const Params& params)
```

Comment:
We seem to have come to a misunderstanding here. This function applies parameterized libraries and should be called accordingly, i.e., `apply_parameterized_gate_library`. One such library with parameters is the `sidb_on_the_fly_gate_library`. 

Think of it this way: the original function here is not called `apply_qca_one_gate_library` or `apply_bestagon_gate_library`. It can apply any (static) gate library. Thus, this function should also be able to handle all parameterized gate libraries.

Reply:
oh sorry, I was too quick here.